tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -822,14 +822,21 @@ void HWCommandQueue::copy_into_user_space(const detail::ReadBufferDescriptor &re
                     uint32_t num_bytes_to_copy;
                     if (remaining_bytes_of_nonaligned_page > 0) {
                         // Case 1: Portion of the page was copied into user buffer on the previous completion queue pop.
-                        num_bytes_to_copy = remaining_bytes_of_nonaligned_page - pad_size_bytes;
-                        remaining_bytes_of_nonaligned_page = 0;
+                        if (remaining_bytes_of_nonaligned_page <= pad_size_bytes) {
+                            num_bytes_to_copy = page_size;
```

Comment:
I had similar fixes put in for my sharding stuff. I think a small corner case left that isn't handled properly here is the handling of pages larger than what you get back out of completion queue, so here we wouldn't just assume we read a full page. I can fix this with my merge

Reply:
yeah, I tried to account for that in the `Case 2: ` section where a portion of the page containing the actual data is split